### DISCUSSION TITLE
universal gpu backend optimization on Linux

### DISCUSSION THREAD

**Discussion by thewh1teagle** [microsoft/onnxruntime#21824] [General]
Is there a general-purpose gpu optimization available for linux? for example, vulkan supports nvidia, amd, and intel gpus. does onnxruntime have a similar universal backend?

**Comment by artyom-beilis** [comment]
Actually I'm interested in building something like that. There is somewhat limited [dlprimitives](https://github.com/artyom-beilis/dlprimitives) backend. But it supports major classification conv nets- I want to continue to expand it - if it is withing onnxruntime or outside.

**Reply by thewh1teagle** [reply]
Interesting. What's your features plans for that comparing to ggml / onnxruntime?
I think that there's no need for more than using Vulkan/CoreML/CPU backends. it's supported on most platforms and CPUs/GPUs these days.

**Reply by artyom-beilis** [reply]
Several things - I orient on GPU rather than CPU. Also there is CPU opencl implementations, the code is far from optimal for them, but GPU is indeed.

Regarding onnxruntime - currently operator set is very small it covers basic convolutional nets but a lot need to be done. I'm intesively working now on pytorch backed but improving ONNX pipeline is planned. That is why I asked about integration with onnxruntime since the project may benefit from its existing goodies (I hope)
